src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -324,14 +335,21 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         prim_ty: PrimitiveType,
         ns: Namespace,
         item_name: Symbol,
-    ) -> Option<(Res, DefId)> {
+    ) -> Vec<(Res, DefId)> {
```

Comment:
Can you return a `PerNS` here instead to enforce at compile time that there's only one result per namespace?

Reply:
I don't get what you mean by "enforce at compile time". We'd need to construct a `PerNS` by calling `prim_ty.impls(tcx)` which is not a constant method and cannot be one since it makes allocations in `simplified_types`. Did you have something else in mind?

Reply:
It's only for primitive types. So normally, this method should never return more than one item. It's possible to create duplicates with trait impls on primitive though, hence why I did this change.